### PR TITLE
Return 201 for successfully created JSON API records

### LINKED ISSUE #496: Successful JSON API POST returns Status Code 200 instead of 201
http://jsonapi.org/format/#crud-creating-responses-201

### PR DESCRIPTION
**Pull request by bravo-kernel**
Status code after successfully created JSON API record should return 201 (now 200). http://jsonapi.org/format/#crud-creating-responses-201.

Closes #496

### CODE REVIEW

**Comment by bravo-kernel** [comment]
I have uploaded the failing integration test to hopefully make it easier to help me out with ticket #513.

**Comment by bravo-kernel** [comment]
This is weird, some Travis builds succeed, others fail. https://travis-ci.org/FriendsOfCake/crud/builds/217468961

**Comment by bravo-kernel** [comment]
nvm, seems the passing tests do not run phpunit 😉

File: tests/TestCase/Controller/JsonApiIntegrationTest.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                'Content-Type' => 'application/vnd.api+json'
+            ]
+        ]);
+        $result = $this->post('/countries', []);
+
+        // These tests do not pass, the expected JSON API validation exception is not thrown
+         $this->assertResponseError();
+         $this->assertResponseCode(422);
+
+        // and this prints the index JSON API instead ?!?
```
Comment (ADmad) [inline tests/TestCase/Controller/JsonApiIntegrationTest.php]: It does so because you haven't connected any RESTful route. You should do so in `setUp()`.

**Comment by bravo-kernel** [comment]
Aha, I assumed it was using 'config/routes.php' in the test App. I will give this a go. Much appreciated!

**Comment by bravo-kernel** [comment]
This is good to merge if the location I chose for the test is acceptable. Happy to move if need be:

```bash
tests/TestCase/Controller/JsonApiIntegrationTest.php 
```

**Comment by jippi** [comment]
seems legit to me

**Pull request merged**